external/webserver/http_client.c

```diff
@@ -167,6 +167,45 @@ int http_client_release(struct http_client_t *client)
 	return HTTP_OK;
 }
 
+int process_trailer_header(char *buf, int buf_len, struct http_req_message *req,
+				struct http_keyvalue_list_t *params,
+				struct http_client_t *client, int sentence_start)
+{
+	char *header_field_value = NULL;
+	int sentence_end;
+	char *entity;
+	int trailter_len;
+
+	/* Process the Trailer */
+	header_field_value = http_keyvalue_list_find(params, "Trailer");
+	if (strncmp(header_field_value, "(null)", 7)) {
+		/* match the Trailer field */
+		if (strncmp(buf + sentence_start, header_field_value, strlen(header_field_value)) == 0) {
+			HTTP_LOGE("Trailer header: %s\n", header_field_value);
+			sentence_end = http_find_first_crlf(buf, buf_len, sentence_start);
+			if (sentence_end < 0) {
+				HTTP_LOGE("Error: Not accord with chunked encoding\n");
+				return -1;
+			}
+
+			/* copy Trailer data */
+			trailter_len = sentence_end - sentence_start;
+			entity = HTTP_MALLOC(trailter_len + 1);
+			if (entity == NULL) {
+				HTTP_LOGE("Error: Fail to alloc memory\n");
+				return -1;
+			}
+			memset(entity, 0, trailter_len + 1);
+			memcpy(entity, buf + sentence_start, trailter_len);
+			req->entity = entity;
+			req->entity_len = trailter_len;
+			http_dispatch_url(client, req);
+		}
+	}
+
+	return 1;
```

Comment:
so it should return 1 even if result of first strncmp is not true? otherwise, you should apply below style
```
if (!strncmp((header_field_value, "(null)", 7)) {
     return -1;
}

if (strncmp(buf + sentence_start, header_field_value, strlen(header_field_value)) == 0) {
....
}
```